packages/sanity/src/core/releases/tool/detail/review/DocumentDiff.tsx

```diff
@@ -0,0 +1,77 @@
+import {diffInput, wrap} from '@sanity/diff'
+import {type ObjectSchemaType, type SanityDocument} from '@sanity/types'
+import {Text} from '@sanity/ui'
+import {useMemo} from 'react'
+import {ChangeResolver, type ObjectDiff} from 'sanity'
+import {DocumentChangeContext} from 'sanity/_singletons'
+
+import {buildChangeList} from '../../../../field/diff/changes/buildChangeList'
+import {ChangesWrapper, FieldWrapper} from './DocumentDiff.styled'
+
+const buildDocumentForDiffInput = (document: SanityDocument) => {
+  // Remove internal fields and undefined values
+  const {_id, _rev, _createdAt, _updatedAt, _type, _version, ...rest} = JSON.parse(
+    JSON.stringify(document),
+  )
+
+  return rest
+}
+
+/**
+ * Compares two documents with the same schema type.
+ * Showing the changes introduced by the document compared to the base document.
+ */
+export function DocumentDiff({
+  baseDocument,
+  document,
+  schemaType,
+}: {
+  baseDocument: SanityDocument | null
+  document: SanityDocument
+  schemaType: ObjectSchemaType
+}) {
+  const {changesList, rootDiff} = useMemo(() => {
+    if (!baseDocument) return {changesList: [], rootDiff: null}
+    const diff = diffInput(
+      wrap(buildDocumentForDiffInput(baseDocument), null),
+      wrap(buildDocumentForDiffInput(document), null),
+    ) as ObjectDiff
+
+    if (!diff.isChanged) return {changesList: [], rootDiff: null}
+    const changeList = buildChangeList(schemaType, diff, [], [], {})
+    return {changesList: changeList, rootDiff: diff}
+  }, [baseDocument, document, schemaType])
+
+  const isChanged = rootDiff?.isChanged || false
```

Comment:
really small nit: I'm having a hard time mapping this /but/ I think that might be best to to use here is the `??` operator? 🤔

Reply:
Wdyt of 
```suggestion
  const isChanged = !!rooitDiff?.isChanged
```

Reply:
Love it :)